File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        
        charSet = set()
        longestSubstring = 0
        left = 0

        for right in range(len(s)):

            while s[right] in charSet:
                charSet.remove(s[left])
                left += 1
            
            charSet.add(s[right])
            longestSubstring = max(longestSubstring, right-left+1)

        return longestSubstring
        

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.java
class Solution {
    public int maxProfit(int[] prices) {
        int left = 0;
        int right = 1;
        int maxProf = 0;

        while(right < prices.length){
            if(prices[left] < prices[right]){
                maxProf = Math.max(maxProf, prices[right] - prices[left]);
            }
            else{
                left = right;
            }
            right++;
        }

        return maxProf;
    }
}
